ipn/serve.go

```diff
@@ -62,6 +78,41 @@ type FunnelConn struct {
 	Src netip.AddrPort
 }
 
+// ServeStreamRequest defines the json request body
+// for the serve stream endpoint
+type ServeStreamRequest struct {
+	// HostPort is the DNS and port of the tailscale
+	// URL.
+	HostPort HostPort `json:",omitempty"`
+
+	// Source is the user's serve destination
+	// such as their localhost server.
```

Comment:
can it have a port? a URL scheme? give some example values.

Reply:
@bradfitz this is the argument as is that is called source and defined with examples here: https://github.com/tailscale/tailscale/blob/50b558de745582873dd89f39ecc14a0d19401c76/cmd/tailscale/cli/serve.go#L43

I can mention that and copy the examples.